hub-chart/templates/pre-delete-hook.yaml

```diff
@@ -0,0 +1,49 @@
+{{- if and (index .Values "kubeflex-operator" "install") (or .Values.ITSes .Values.WDSes ) }}
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: "{{ .Release.Name }}-delete-cps"
+  namespace: kubeflex-system
+  labels:
+    app.kubernetes.io/managed-by: {{ .Release.Service | quote }}
+    app.kubernetes.io/instance: {{ .Release.Name | quote }}
+    app.kubernetes.io/version: {{ .Chart.AppVersion }}
+    helm.sh/chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+  annotations:
+    # This is what defines this resource as a hook. Without this line, the
+    # job is considered part of the release.
+    "helm.sh/hook": pre-delete
+    "helm.sh/hook-weight": "-5"
+    "helm.sh/hook-delete-policy": hook-succeeded
+spec:
+  template:
+    metadata:

```

Comment:
I notice that none of the example Job objects in https://kubernetes.io/docs/concepts/workloads/controllers/job have a `spec.template.metadata` section. Do we need to/can we/should we specify the pod name? The pod labels?

Reply:
@MikeSpreitzer is this specific for the pre-delete-hook policy? I can change it if you let me know what you want.

Reply:
I thought that this is the only Job YAML with `spec.template.metadata`. I am not yet asserting something, I am trying to understand. Is there a specific need to specify the `spec.template.metadata` in this case? I will say that if it is not needed then perhaps it should not be there. In fact, I suspect there may be some confusion here. The Job controller usually takes care of the Pod names.  Following is an example of a Pod from the Job that invokes `clusteradm init`; the Job is named "ocm".

```
mspreitz@mjs13 ocm-status-addon % kubectl --context kind-kubeflex get pods -n its1-system ocm-tlhrp -o yaml
apiVersion: v1
kind: Pod
metadata:
  generateName: ocm-
  labels:
    batch.kubernetes.io/controller-uid: e6edf60f-e1eb-4cbb-88ee-5910cd30245f
    batch.kubernetes.io/job-name: ocm
    controller-uid: e6edf60f-e1eb-4cbb-88ee-5910cd30245f
    job-name: ocm
  name: ocm-tlhrp
  namespace: its1-system
  ownerReferences:
  - apiVersion: batch/v1
    blockOwnerDeletion: true
    controller: true
    kind: Job
    name: ocm
    uid: e6edf60f-e1eb-4cbb-88ee-5910cd30245f
...
```

Reply:
I wonder if there is some confusion here between the Job's `metadata` and that of the PodSpec inside the Job (`spec.template.metadata`).